src/modules/product/interface/product.routes.ts

```
@@ -30,6 +30,31 @@ productRouter.get('/store/:storeId', (req, res, next) =>
+);
+
+// Create new product with image
+productRouter.post('/', (req, res, next) => {
+  logger.info('Product creation request received:', {
+    body: req.body,
+    files: req.files,
+    file: req.file,
+    headers: req.headers,
+  });
+
+  handleProductUpload(req, res, next, (req: Request, res: Response, next: NextFunction) => {
```

ESLint

[error] 'handleProductUpload' was used before it was defined.

(@typescript-eslint/no-use-before-define)